random-distribution-helpers.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"strings"
+	"time"
+)
+
+type RandomDistribution interface {
+	Sample() time.Duration
+	String() string
+}
+
+var _ RandomDistribution = RandomConstDistribution(nil)
+
+type RandomConstDistribution struct {
+	value time.Duration
+}
+
+func (constDist RandomConstDistribution) Sample() time.Duration {
+	return constDist.value
+}
+
+func (constDist RandomConstDistribution) String() string {
+	return fmt.Sprintf("Constant value: %d.", constDist.value)
+}
+
+var _ RandomDistribution = RandomEqualDistribution(nil)
+
+type RandomEqualDistribution struct {
+	min time.Duration
+	max time.Duration
+}
+
+func (equalDist RandomEqualDistribution) Sample() time.Duration {
+	return time.Duration(rand.Int63n(int64(equalDist.max) - int64(equalDist.min)) + int64(equalDist.min))
+}
+
+func (equalDist RandomEqualDistribution) String() string {
+	return fmt.Sprintf("Equal distribution between %d and %d.", equalDist.min, equalDist.max)
+}
+
+var _ RandomDistribution = RandomNormalDistribution(nil)
+
+type RandomNormalDistribution struct {
+	mu    time.Duration
+	sigma time.Duration
+}
+
+func (normDist RandomNormalDistribution) Sample() time.Duration {
+	value := math.Round(rand.NormFloat64() * float64(normDist.sigma) + float64(normDist.mu))
+	return time.Duration(value)
+}
+
+func (normDist RandomNormalDistribution) String() string {
+	return fmt.Sprintf("Normal distribution with mean %d and standard deviation %d.", normDist.mu, normDist.sigma)
+}
+
+type RandomDistributionSampler struct {
+	distribution	RandomDistribution
+}
+
+func (distSampler RandomDistributionSampler) String() string {
+	return distSampler.distribution.String()
+}
+
+func (distSampler RandomDistributionSampler) Set(value string) error {
+	formatErr := "Invalid random argument format. Please use format [const|equal|norm]:[param1, param2,...]. Reason: %v"
+	if len(value) == 0 || !strings.Contains(value, ":") {
+		return fmt.Errorf(formatErr, "Distribution type and parameters must be devided by ':'.")
+	}
+	typeAndParams := strings.Split(value, ":")
+	if len(typeAndParams) != 2 {
+		return fmt.Errorf(formatErr, "Missing distribution parameters.")
```

Comment:
The two error messages are redundant: the first if condition is covered by the second one. strings.Split() will return an array with a single element if the input string is empty or does not contain a colon.